test-data/unit/check-functools.test

```diff
@@ -658,3 +658,71 @@ def f(x: P):
     # TODO: but this is incorrect, predating the functools.partial plugin
     reveal_type(partial(x, "a")())  # N: Revealed type is "builtins.int"
 [builtins fixtures/tuple.pyi]
+
+[case testFunctoolsPartialTypeVarErasure]
+from typing import Callable, TypeVar, Union
+from typing_extensions import ParamSpec, TypeVarTuple, Unpack
+from functools import partial
+
+def use_int_callable(x: Callable[[int], int]) -> None:
+    pass
+def use_func_callable(
+    x: Callable[
+        [Callable[[int], None]],
+        Callable[[int], None],
+    ],
+) -> None:
+    pass
+
+Tc = TypeVar("Tc", int, str)
+Tb = TypeVar("Tb", bound=Union[int, str])
+P = ParamSpec("P")
+Ts = TypeVarTuple("Ts")
+
+def func_b(a: Tb, b: str) -> Tb:
+    return a
+def func_c(a: Tc, b: str) -> Tc:
+    return a
+
+def func_fn(fn: Callable[P, Tc], b: str) -> Callable[P, Tc]:
+    return fn
+def func_fn_unpack(fn: Callable[[Unpack[Ts]], Tc], b: str) -> Callable[[Unpack[Ts]], Tc]:
+    return fn
+
+# We should not leak stray typevars that aren't in scope:
+use_int_callable(partial(func_b, b=""))
+use_func_callable(partial(func_b, b=""))
```

Comment:
Why does this pass? I assume the erased return type is `Union[int, str]` and not `Never`...

Also could you add some `reveal_type` like `use_func_callable(reveal_type(partial(func_b, b="")))`

Reply:
The erased return type is `Any`. `erase_typevars` replaces them with `Any` (`special_form`), not with upper bounds.

Adding `reveal_type` is a good idea, let's show the inferred generics explicitly. I don't even need `use_*_callable` then...

Reply:
I think you should still have the `use_*_callable` to ensure things still work, at least in a few spots -- printing correctly doesn't necessarily mean the logic is right, e.g. maybe the plugin could have messed up something (?).

Reply:
And `Any` is really what we want. Replacing tvars with their upper bounds will cause another bunch of false positives whenever smth like `def fn[T: str | int](x: T) -> T` (hope I didn't mess up this ugly syntax) is used with partial.